crypto/hmac/hmac.c

```diff
@@ -35,6 +35,10 @@ int HMAC_Init_ex(HMAC_CTX *ctx, const void *key, int len,
         return 0;
     }
 
+    /* We currently don't support shake128 and shake256 with HMAC */
+    if ((EVP_MD_meth_get_flags(md) & EVP_MD_FLAG_XOF) != 0)
+        return 0;
```

Comment:
I agree with @kroeckx  that "currently" should be dropped because it is misleading. How about the following alternative formulation?

```C
/*
 * The HMAC construction is not allowed  to be used with the
 * extendable-output functions (XOF) shake128 and shake256.
 */
```


_(BTW: @tiran you could have told us this [long time ago](https://bugs.python.org/issue28384)  ;-) )_